=== modified file 'Bugzilla.pm'
--- Bugzilla.pm	2011-12-19 18:47:35 +0000
+++ Bugzilla.pm	2012-01-19 04:59:23 +0000
@@ -81,7 +81,7 @@
 # Global Code
 #####################################################################
 
-# $::SIG{__DIE__} = i_am_cgi() ? \&CGI::Carp::confess : \&Carp::confess;
+$::SIG{__DIE__} = i_am_cgi() ? \&CGI::Carp::confess : \&Carp::confess;
 
 # Note that this is a raw subroutine, not a method, so $class isn't available.
 sub init_page {

