=== modified file 'template/en/default/list/list.html.tmpl'
--- template/en/default/list/list.html.tmpl 2013-01-16 18:05:22 +0000
+++ template/en/default/list/list.html.tmpl 2013-04-26 06:04:57 +0000
@@ -74,7 +74,7 @@
     <div class="bz_query_debug">
       <p>Total execution time: [% query_time FILTER html %] seconds</p>
       [% FOREACH query = queries %]
-        <p>[% query.sql FILTER html %]</p>
+        <pre>[% query.sql FILTER html %]</pre>
         <p>Execution time: [% query.time FILTER html %] seconds</p>
         [% IF query.explain %]
           <pre>[% query.explain FILTER html %]</pre>

