=== modified file 'buglist.cgi'
--- buglist.cgi	2012-06-27 16:12:10 +0000
+++ buglist.cgi	2012-07-04 06:40:49 +0000
@@ -788,6 +788,10 @@
 if ($cgi->param('debug')) {
     $vars->{'debug'} = 1;
     $vars->{'query'} = $query;
+
+    use Mysql::PrettyPrinter;
+    $vars->{'query'} = Mysql::PrettyPrinter->format(sql => $query);
+
     # Explains are limited to admins because you could use them to figure
     # out how many hidden bugs are in a particular product (by doing
     # searches and looking at the number of rows the explain says it's

=== modified file 'template/en/default/list/list.html.tmpl'
--- template/en/default/list/list.html.tmpl	2012-02-25 14:08:23 +0000
+++ template/en/default/list/list.html.tmpl	2012-07-04 06:41:52 +0000
@@ -58,7 +58,7 @@
   </span>
 
   [% IF debug %]
-    <p class="bz_query">[% query FILTER html %]</p>
+    <pre style="text-align:left">[% query FILTER html %]</pre>
     [% IF query_explain.defined %]
       <pre class="bz_query_explain">[% query_explain FILTER html %]</pre>
     [% END %]

